storagetransfer/quickstart_test.go

```diff
@@ -0,0 +1,111 @@
+package storagetransfer
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+	"testing"
+
+	"cloud.google.com/go/iam"
+	"cloud.google.com/go/storage"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	storagetransfer "cloud.google.com/go/storagetransfer/apiv1"
+	storagetransferpb "google.golang.org/genproto/googleapis/storagetransfer/v1"
+)
+
+var str *storage.Client
+var sts *storagetransfer.Client
+
+func TestQuickstart(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	ctx := context.Background()
+
+	c, err := storage.NewClient(ctx)
+	if err != nil {
+		log.Fatalf("storage.NewClient: %v", err)
+	}
+	str = c
+	defer str.Close()
+
+	s, err := storagetransfer.NewClient(ctx)
+	if err != nil {
+		log.Fatalf("storagetransfer.NewClient: %v", err)
+	}
+	sts = s
+	defer sts.Close()
+
+	sinkBucketName, err := testutil.CreateTestBucket(ctx, t, str, tc.ProjectID, "sts-go-sink")
+	if err != nil {
+		t.Fatalf("Bucket creation failed: %v", err)
+	}
+	defer testutil.DeleteBucketIfExists(ctx, str, sinkBucketName)
+
+	sourceBucketName, err := testutil.CreateTestBucket(ctx, t, str, tc.ProjectID, "sts-go-source")
+	if err != nil {
+		t.Fatalf("Bucket creation failed: %v", err)
+	}
+	defer testutil.DeleteBucketIfExists(ctx, str, sourceBucketName)
+
+	grantStsPermissions(sinkBucketName, tc.ProjectID)
+	grantStsPermissions(sourceBucketName, tc.ProjectID)
+
+	buf := new(bytes.Buffer)
+	if err := quickstart(buf, tc.ProjectID, sourceBucketName, sinkBucketName); err != nil {
+		t.Errorf("quickstart: %#v", err)
+	}
+
+	got := buf.String()
+	if want := "transferJobs/"; !strings.Contains(got, want) {
+		t.Errorf("got %q, want %q", got, want)
+	}
+
+	re := regexp.MustCompile("transferJobs/\\d+")
+	name := re.FindString(got)
+
+	tj := &storagetransferpb.TransferJob{
+		Name: name,
+		Status: storagetransferpb.TransferJob_DELETED,
+	}
+	sts.UpdateTransferJob(ctx, &storagetransferpb.UpdateTransferJobRequest{
+		JobName: name,
+		ProjectId: tc.ProjectID,
+		TransferJob: tj,
+	})
+}
+
+func grantStsPermissions(bucketName string, projectID string) {
+	ctx := context.Background()
+
+	req := &storagetransferpb.GetGoogleServiceAccountRequest{
+		ProjectId: projectID,
+	}
+
+	resp, err := sts.GetGoogleServiceAccount(ctx, req)
```

Comment:
Possibly totally off-base but-- should this code really be in a sample as well, rather than just in the test? Do the docs make it clear how to give these permissions to the STS service account?

Reply:
The docs explain how to do it in the console, but not with code. I agree that it should be a sample and it's something I want to add and will propose at some point, just not sure how to go about it yet since it's done entirely with the GCS client and not with the STS client, so there has to be some explainer on how to depend on the GCS client as well, I'm not sure if there's other examples of docs in one client having samples for a different client. Either way, it's not going to be part of the this stage